File: src/main/java/com/severett/kotlinintro/model/OfferedService.java
package com.severett.kotlinintro.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

import static jakarta.persistence.GenerationType.IDENTITY;

@Builder(toBuilder = true)
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class OfferedService {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    private int id;
    private String name;
    private BigDecimal price;
    private BigDecimal discount;

    @PrePersist
    public void prePersist() {
        if (discount == null) discount = BigDecimal.ZERO;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof OfferedService that)) return false;
        return id == that.id;
    }

    @Override
    public int hashCode() {
        return 0;
    }
}

File: src/main/kotlin/com/severett/kotlinintro/model/OfferedService.kt
package com.severett.kotlinintro.model

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.math.BigDecimal

@Entity
data class OfferedService(
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,
    val name: String = "",
    val price: BigDecimal = BigDecimal.ZERO,
    val discount: BigDecimal = BigDecimal.ZERO
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is OfferedService) return false

        return id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()
}
